### PR TITLE
Upgrade to Terraform SDK v2

### PR DESCRIPTION
**Pull request by eravin-ns1**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
+## 1.14.x (???)
```
Comment (Zach-Johnson) [inline CHANGELOG.md:1]: Fix header

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+ENHANCEMENTS
+
+* Upgraded to Terraform SDK 2.24.1. Users of Pulsar will need to make minor changes in their resource files, see below.
+
+INCOMPATIBILITIES WITH PREVIOUS VERSIONS
+
+* The `ns1_application` resource attributes `config`, `default_config` and `blended_metric_weights` are now blocks, with only one item permitted. This is due to an SDK 2.x restriction on nested structures. Existing resource files will need to be edited to remove the equals sign in the declarations of the affected stanzas, for example:
```
Comment (Zach-Johnson) [inline CHANGELOG.md:8]: this seems like it could warrant a major version bump

**Review by Zach-Johnson — COMMENTED** [review]
dropped a couple of notes

File: CHANGELOG.md (hunk 2 above)
Comment (eravin-ns1) [reply CHANGELOG.md:8]: Done.

**Review by Zach-Johnson — APPROVED** [review]

**Pull request merged**